test/middleware/decrypt_card_data_test.js

```diff
@@ -0,0 +1,211 @@
+const path = require('path')
+const sinon = require('sinon')
+const proxyquire = require('proxyquire')
+const { expect } = require('chai')
+
+describe('decryptCardData middleware', function () {
+  let next
+  beforeEach(function () {
+    next = sinon.stub()
+    next.promise = new Promise(resolve => next.callsFake(resolve))
```

Comment:
This doesn't handle promise rejections when `next` is called with an error, so some of the tests are probably not working correctly.

Reply:
Fixed in 1e0047d9